tests/unit/database_tests.py

```diff
@@ -92,13 +93,36 @@ def test_retrieve_creds(self):
 
     def test_exists(self):
         """
-        Test database exists fucntionality
+        Test database exists functionality
         """
         self.assertTrue(self.db.exists())
         # Construct a database object that does not exist remotely
         fake_db = self.client._DATABASE_CLASS(self.client, 'no-such-db')
         self.assertFalse(fake_db.exists())
 
+    def test_exists_raises_bad_request(self):
+        """
+        Test database exists raises a 400 bad request error
+        """
+        if os.environ.get('RUN_CLOUDANT_TESTS') is None:
+            with self.assertRaises(requests.HTTPError) as cm:
+                db = self.client._DATABASE_CLASS(self.client, '/')
+                db.exists()
+            err = cm.exception
+            self.assertEqual(err.response.status_code, 400)
+        else:
+            # Mock HTTP bad request when running against Cloudant
+            mock_resp = mock.Mock()
+            mock_resp.raise_for_status.side_effect = requests.HTTPError(
+                mock.Mock(), 'bad request')
+            mock_resp.raise_for_status.response = mock.Mock(code=400)
+            # mock_session.head = mock.Mock(return_value=mock_resp)
+            self.client.r_session.head = mock.Mock(return_value=mock_resp)
+            with self.assertRaises(requests.HTTPError):
+                self.db.exists()
+            self.assertTrue(self.client.r_session.head.called)
+            self.assertEqual(1, mock_resp.raise_for_status.call_count)
+
```

Comment:
My suggestion here would be to have one test that functions for both CouchDB and Cloudant that mocks the response code rather than introduce this `if` logic in a unit test.  Also I would not call this a "bad request" test because it implies that this is only testing a 400 return code which if you change this to a mocked test for both CouchDB and Cloudant as I suggested, this test would cover all raised HTTPErrors.  So... `test_exists_raises_httperror`.